cloudify_aws/ec2/resources/networkacl.py

```diff
@@ -181,51 +169,61 @@ def attach(ctx, iface, resource_config, **_):
     # only use a relationship. But this code used to be here so
     # we are stuck with it.
     subnet_id = params.get(SUBNET_ID)
-    new_network_acl_association_ids = []
+    network_acl_assoc_prop = {}
     subnet_ids = utils.find_ids_of_rels_by_node_type(
         ctx.instance, SUBNET_TYPE)
     if subnet_id:
         subnet_ids.append(subnet_id)
-    for subnet in subnet_ids:
-        network_acl_associations = \
-            iface.get_properties_by_filter(
-                ASSOCIATION_SUBNET_ID, subnet)
-        network_acl_association_id = \
-            network_acl_associations.get(
-                'Associations')[0].get(
-                'NetworkAclAssociationId')
-        params.update({ASSOCIATION_ID: network_acl_association_id})
-        default_acl_id = \
-            network_acl_associations.get(
-                'Associations')[0].get('NetworkAclId')
-        ctx.instance.runtime_properties['default_acl_id'] = \
-            default_acl_id
-
-        # # Actually attach the resources
-        new_network_acl_association_list = iface.attach(params)
-        new_network_acl_association_id = \
-            new_network_acl_association_list.get(
-                'NewAssociationId')
-        new_network_acl_association_ids.append(
-            new_network_acl_association_id)
-
-    if new_network_acl_association_ids and \
-            isinstance(new_network_acl_association_ids, list):
-        ctx.instance.runtime_properties['association_ids'] = \
-            new_network_acl_association_ids
-        ctx.instance.runtime_properties['association_id'] = \
-            new_network_acl_association_ids[0]
+    try:
+        vpc_id = utils.find_ids_of_rels_by_node_type(
+            ctx.instance, VPC_TYPE)[0]
+    except IndexError:
+        vpc_id = None
+    acl_associations = iface.get_network_acls()
+    for acl in acl_associations['NetworkAcls']:
+        if not acl[VPC_ID] == vpc_id and len(acl['Associations']) < 1:
```

Comment:
I think re-writing this would be more readable
```
if acl[VPC_ID] != vpc_id and len(acl['Associations']) < 1:
``` 

Moreover,  What if the ``iface.get_network_acls()`` return None or empty list , you will end up wit error ?

Reply:
👍